bot/exts/evergreen/status_codes.py

```diff
@@ -20,62 +26,53 @@ class HTTPStatusCodes(commands.Cog):
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 404:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text="Inputted status code does not exist.")
-
-        except NotImplementedError:
-            embed.set_footer(text="Inputted status code is not implemented by http.cat yet.")
-
-        finally:
-            await ctx.send(embed=embed)
+        """Send a cat version of the requested HTTP status code."""
+        if code in range(100, 600):
+            await self.build_embed(url=HTTP_CAT_URL.format(code=code), ctx=ctx, code=code)
+            return
+        await ctx.send(embed=ERROR_LENGTH_EMBED)
 
     @http_status_group.command(name="dog")
     async def http_dog(self, ctx: commands.Context, code: int) -> None:
-        """Sends an embed with an image of a dog, portraying the status code."""
-        # These codes aren't server-friendly.
-        if code in (304, 422):
-            await self.http_cat(ctx, code)
+        """Send a dog version of the requested HTTP status code."""
+        if code in range(100, 600):
+            await self.build_embed(url=HTTP_DOG_URL.format(code=code), ctx=ctx, code=code)
             return
-
-        embed = discord.Embed(title=f"**Status: {code}**")
-        url = HTTP_DOG_URL.format(code=code)
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 302:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text="Inputted status code does not exist.")
-
-        except NotImplementedError:
-            embed.set_footer(text="Inputted status code is not implemented by httpstatusdogs.com yet.")
-
-        finally:
-            await ctx.send(embed=embed)
+        await ctx.send(embed=ERROR_LENGTH_EMBED)
+
+    async def build_embed(self, url: str, ctx: commands.Context, code: int) -> None:
+        """Attempt to build and dispatch embed. Append error message instead if something goes wrong."""
+        async with self.bot.http_session.get(url, allow_redirects=False) as response:
+            if response.status in range(200, 300):
+                await ctx.send(
+                    embed=discord.Embed(
+                        title=STATUS_TEMPLATE.format(code=code)
+                    ).set_image(url=url)
+                )
+            elif response.status in (302, 404):  # dog URL returns 302 instead of 404
+                await ctx.send(
+                    embed=discord.Embed(
+                        title=ERR_404.format(code=code)
+                    ).set_image(url=url)
```

Comment:
Whilst the cat api returns the 404 image if the result isn't found, meaning the URL can be used anyway, the dog api does not, meaning discord will try to use an invalid URL and fail to load the image. Explicitly using the relevant 404 image would be a way to fix this. Example to test this is `.status dog 478`

Reply:
Would this be the correct behavior if the status code is not found?

I hard-coded in the 404 pictures for the dog and cat urls if there is a 404 error. 

![image](https://user-images.githubusercontent.com/42116429/131663526-52bf332e-df53-40c5-9063-15e7e9bf40ce.png)

Reply:
That looks good to me 👍